packages/wp-dashboard/src/components/editorSettings/customFonts/index.js

```diff
@@ -0,0 +1,325 @@
+  const [showDialog, setShowDialog] = useState(false);
+  const [toDelete, setToDelete] = useState(null);
+  const canSave = !inputError;
+
+  const handleUpdateFontUrl = useCallback((event) => {
+    const { value } = event.target;
+    setFontUrl(value);
+
+    if (value.length === 0 || isValidUrl(withProtocol(value))) {
+      setInputError('');
+
+      return;
+    }
+
+    setInputError(TEXT.INPUT_ERROR);
+  }, []);
+
+  const handleDelete = useCallback(async () => {
+    await deleteCustomFont(toDelete);
+    await fetchCustomFonts();
+    setToDelete(null);
+    setShowDialog(false);
+  }, [toDelete, deleteCustomFont, fetchCustomFonts]);
+
+  const handleOnSave = useCallback(async () => {
+    if (canSave) {
+      const urlWithProtocol = withProtocol(fontUrl);
+      try {
+        await fetch(urlWithProtocol, {
+          method: 'HEAD',
+        });
+      } catch (err) {
+        setInputError(
+          __(
+            'Please ensure correct CORS settings for allowing font usage on this site.',
+            'web-stories'
+          )
+        );
+      }
+      try {
+        trackEvent('add_custom_font', {
+          url: urlWithProtocol,
+        });
+        const fontData = await getFontDataFromUrl(urlWithProtocol);
+        if (!fontData.family) {
+          setInputError(
+            __('Something went wrong, please try again.', 'web-stories')
+          );
+        } else {
+          await addCustomFont({ ...fontData, url: urlWithProtocol });
+          await fetchCustomFonts();
+          setFontUrl('');
+        }
+      } catch (err) {
+        setInputError(
+          __(
+            'Getting font data failed, please ensure the URL points directly to a .otf, .ttf or .woff file.',
+            'web-stories'
+          )
+        );
+      }
+    }
+  }, [addCustomFont, fetchCustomFonts, canSave, fontUrl]);
+
+  const handleOnKeyDown = useCallback(
+    (e) => {
+      if (e.key === 'Enter') {
+        e.preventDefault();
+        handleOnSave();
+      }
+    },
+    [handleOnSave]
+  );
+
+  return (
+    <SettingForm onSubmit={(e) => e.preventDefault()}>
+      <div>
+        <SettingHeading as="h3">{TEXT.SECTION_HEADING}</SettingHeading>
+        <SettingSubheading size={THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL}>
+          {TEXT.ADD_CONTEXT}
+        </SettingSubheading>
+        <SettingSubheading size={THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL}>
+          {TEXT.REMOVAL}
+        </SettingSubheading>
+      </div>
+      <InputsWrapper>
+        <InlineForm>
+          <SettingsTextInput
+            id="insertFontUrl"
+            value={fontUrl}
+            onChange={handleUpdateFontUrl}
+            onKeyDown={handleOnKeyDown}
+            hasError={Boolean(inputError)}
+            hint={inputError}
+            label={TEXT.LABEL}
+          />
+          <AddButton
+            type={BUTTON_TYPES.PRIMARY}
+            size={BUTTON_SIZES.SMALL}
+            disabled={inputError}
```

Comment:
Also disable when the input's empty for consistency with the Google analytics 'save' button

Reply:
Ah, good point, will do, thanks!

I also saw the button type inconsistencies in UX for the button, also, the Delete modal is light themed whereas in some other places I saw it dark themed in the Dashboard. I'll follow up with Amy about it to see which way to go, probably would be good to have just one.

Reply:
good catch on the modal - it should be inverted to the theme used so since dashboard is light theme the modals are dark themed.